poetry/console/commands/add.py

```diff
@@ -173,6 +172,7 @@ def handle(self):
 
             status = self._installer.run()
         except BaseException:
+            # Catch all exception including KeyboardInterrupt in order to restore the original content
```

Comment:
```suggestion
            # Using BaseException here as some exceptions, eg: KeyboardInterrupt, do not inherit from Exception
```

Sorry, should have been more specific about what I had in mind.